src/test/java/wolox/training/BookControllerTest.java

```diff
@@ -0,0 +1,118 @@
+package wolox.training;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.ObjectWriter;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import java.nio.charset.Charset;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers;
+import org.springframework.web.util.NestedServletException;
+import wolox.training.controllers.BookController;
+import wolox.training.exceptions.BookIdMismatchException;
+import wolox.training.exceptions.BookNotFoundException;
+import wolox.training.models.Book;
+import wolox.training.repositories.BookRepository;
+import wolox.training.utils.Constants;
+
+@RunWith(SpringRunner.class)
+@WebMvcTest(BookController.class)
+public class BookControllerTest {
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @MockBean
+    private BookRepository bookRepository;
+
+    private Book testBook;
+
+    @Before
+    public void setUp(){
+        testBook = new Book();
+        testBook.setGenre("Fantasy");
+        testBook.setAuthor("Juan Camilo");
+        testBook.setImage("image.jpg");
+        testBook.setTitle("Libro test");
+        testBook.setSubtitle("test");
+        testBook.setPublisher("Tesstpublisher");
+        testBook.setPages(100);
+        testBook.setYear("2020");
+        testBook.setIsbn("12345678");
+    }
```

Comment:
you can extract this to a another class like `BookFactory` and use in multiple test, for example in BookRepositoryTest, so as not to repeat code later. 
For example `testBook = new BookFactory().build()`